<!-- -->
<launch>
	<arg name="magabot_port" default="/dev/ttyMagabot"/>
	<arg name="inertial_port" default="/dev/ttyMagabotInertial"/>
	<arg name="laser_port" default="/dev/ttyURG0"/>

	<node name ="magabot_node" pkg="magabot" type="magabot_node" output="screen"> 
		<param name="port" value="$(arg magabot_port)" />
		<param name="port_inertial" type="string" value="$(arg inertial_port)" />
		<param name="use_imu" value="true"/>
	</node>

    <arg name="vel_linear_max" default="0.2"/>
	<arg name="vel_angular_max" default="0.4"/>
	<node pkg="magabot" name="joysticknode" type="joystick_node">
		<param name="vel_linear_max" value="$(arg vel_linear_max)" />
		<param name="vel_ang_max" value="$(arg vel_angular_max)" />
	</node>

	<node pkg="joy" name="joy_node" type="joy_node"/>

<!-- LASER -->
	<arg name="rplidar" default="false"/>
	<node name="urg" pkg="urg_node" type="urg_node" respawn="false" output="screen" unless="$(arg rplidar)">
		<param name="frame_id" value="laser" />
		<param name="synchronize_time" type="bool" value="false"/>
		<param name="serial_port" type="string" value="$(arg laser_port)" />
		<param name="publish_intensity" type="bool" value="false"/>
	</node>
	<include file="$(find rplidar_ros)/launch/rplidar_a2m8.launch" if="$(arg rplidar)"/>

<!-- Static TFs a bit over 10Hz: -->
	<arg name="yaw" default="0"/>
	<node pkg="tf" type="static_transform_publisher" name="static_laser_tf" args="0.10 0 0 $(arg yaw) 0 0 base_link laser 99" />

</launch>
